src/www/ui_tests/api/Models/DeciderTest.php

```diff
@@ -11,116 +11,131 @@
  */
 
 namespace Fossology\UI\Api\Test\Models;
-
-use Fossology\UI\Api\Models\Decider;
 use Fossology\UI\Api\Models\ApiVersion;
+use Fossology\UI\Api\Models\Decider;
 
 /**
  * @class DeciderTest
  * @brief Tests for Decider model
  */
 class DeciderTest extends \PHPUnit\Framework\TestCase
 {
+
+  private $decider;
+
   /**
-   * @test
-   * -# Test for Decider::setUsingArray() when $version is V1
-   * -# Create a test array and pass to setUsingArray() on test object
-   * -# Check if the object is updated
+   * @brief Setup method to initialize the Decider object
    */
-  public function testSetUsingArrayV1()
+  public function setUp(): void
   {
-    $this->testSetUsingArray(ApiVersion::V1);
+    $this->decider = new Decider();
   }
 
   /**
    * @test
-   * -# Test for Decider::setUsingArray() when $version is V2
-   * -# Create a test array and pass to setUsingArray() on test object
-   * -# Check if the object is updated
+   * @brief Test for Decider::setUsingArray()
+   * - Create a test array and pass to setUsingArray() on test object
+   * - Check if the object is updated
    */
-  public function testSetUsingArrayV2()
+  public function testSetUsingArray()
   {
-    $this->testSetUsingArray(ApiVersion::V2);
+    $deciderArray = [
+      "nomos_monk" => true,
+      "bulk_reused" => false,
+      "ojo_decider" => (1==1)
+    ];
+
+    $expectedObject = new Decider(true, false, false, true);
+
+    $actualObject = new Decider();
+    $actualObject->setUsingArray($deciderArray);
+
+    $this->assertEquals($expectedObject, $actualObject);
   }
 
   /**
-   * @param $version version to test
-   * @return void
-   * -# Test for Decider::setUsingArray() to check if the object is updated
+   * Provides test data and an instance of the Decider class.
+   * @return array An associative array containing test data and an Agent object.
    */
-  private function testSetUsingArray($version)
+  public function getDeciderInfo($version=ApiVersion::V2)
   {
-    if ($version == ApiVersion::V1) {
-      $deciderArray = [
-        "nomos_monk" => true,
+    if($version==ApiVersion::V1){
+      $deciderInfo = [
+        "nomos_monk"  => true,
         "bulk_reused" => false,
-        "ojo_decider" => (1==1)
+        "new_scanner" => false,
+        "ojo_decider" => true
       ];
-    } else {
-      $deciderArray = [
-        "nomosMonk" => true,
+    }else{
+      $deciderInfo = [
+        "nomosMonk"  => true,
         "bulkReused" => false,
-        "ojoDecider" => (1==1)
+        "newScanner" => false,
+        "ojoDecider" => true
```

Comment:
No changes are needed as this test already ran successfully for both V1 and V2. If we already have both version tests explicitly defined, I don't think we should remove them until V1 is completely deprecated.